crates/primitives/src/checkpoints.rs

```diff
@@ -123,6 +123,26 @@ pub struct StorageHashingCheckpoint {
     pub to: u64,
 }
 
+/// Saves the progress of Headers stage.
+#[main_codec]
+#[derive(Debug, Default, PartialEq, Eq, Clone, Copy)]
+pub struct HeadersCheckpoint {
```

Comment:
Just a general thought here, this can be generalized as just a block checkpoint, these stages also use this exact mechanism:

- Bodies stage
- Execution stage

So this doesn't need to be header specific